### PR TITLE
chore: separate formatting and linting scripts

### PR DESCRIPTION
**Pull request by wincent**
Adds these scripts to package.json, in addition to "format" and "format:check" which we already had:

- "lint": reports lints but doesn't fix them.
- "lint:fix": reports lints and fixes autofixable lints.

An earlier version of this commit in the PR had a "fix" script that was equivalent to "format" followed by "lint:fix" but we're temporarily backing that out until we come up with a better name.

Test plan: Run all the new scripts; see "format" fix formatting issues in one file, and "lint" report these issues which I then fixed:

    packages/liferay-npm-scripts/src/index.js
      61:4  error  Unexpected console statement  no-console

    packages/liferay-npm-scripts/src/scripts/eject.js
      36:33  error  Unnecessary escape character: \"  no-useless-escape
      38:5   error  Unnecessary escape character: \"  no-useless-escape

    packages/liferay-npm-scripts/src/utils/generate-soy-dependencies.js
      28:18  error  Empty block statement  no-empty

    packages/liferay-npm-scripts/src/utils/get-merged-config.js
      23:4  error  Unreachable code              no-unreachable
      31:4  error  Unreachable code              no-unreachable
      39:4  error  Unreachable code              no-unreachable
      50:4  error  Unreachable code              no-unreachable
      52:4  error  Unexpected console statement  no-console

    ✖ 9 problems (9 errors, 0 warnings)

Closes: https://github.com/liferay/liferay-npm-tools/issues/32

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,10 +3,14 @@
...
 	},
 	"scripts": {
-		"format": "prettier-eslint \"./**/*.js\" --write",
-		"lerna": "npm install && lerna bootstrap"
+		"fix": "npm run format && npm run lint:fix",
+		"format:check": "prettier --list-different '**/*.js'",
+		"format": "prettier --write '**/*.js'",
+		"lerna": "npm install && lerna bootstrap",
+		"lint": "eslint '**/*.js'",
+		"lint:fix": "eslint --fix '**/*.js'"
```
Comment (bryceosterhaus) [inline package.json]: How do you feel about changing the names up a bit? Using `format`, `fix`, and `lint:fix` are all a bit confusing without looking at what they actually do, especially `format` and `fix`.

I'm thinking something like 
```json
{
	"format": "npm run prettier && npm run lint:fix",
	"prettier:check": "prettier --list-different '**/*.js'",
	"prettier": "prettier --write '**/*.js'",
	"lerna": "npm install && lerna bootstrap",
	"lint": "eslint '**/*.js'",
	"lint:fix": "eslint --fix '**/*.js'"
}
```

Comment (wincent) [reply package.json]: I'm open to changing the names but I can give you some insight into why they are named they are:

- "format" and "format:check" instead of "prettier" and "prettier:check" in order to say what they do instead of how they do it (an implementation detail); ie. the same reason that we have "test" script that runs `jest` instead of creating a "jest" one. I know it's unlikely we'll ever use anything but Prettier for formatting, but certainly in the case of testing and linting we have legacy projects that use other tools (Karma, Mocha etc).
- "lint" and "lint:fix" for similar reasons.

So I think those are good. "fix", however, is the worst name: I only added it because you wanted one command that did both linting and fixing. I don't think "format" is a good name for that task, because the whole point of this change is to decoupling the concerns of formatting and linting. Do you have any other suggestions for what a good name for this "fix all the things task"?

**Comment by wincent** [comment]
#43 will conflict with this, so I'll rebase/change/update this after that goes in.

File: package.json (hunk 1 above)
Comment (bryceosterhaus) [reply package.json]: Ah I see, I had figured you wanted to go the route of "say what they do instead of how they do it", which I am good with, I think it was the "fix" one that through me off when it is coupled with the rest.

Since the "fix" script was added for my concern of doing it all in one command, I'd say just go ahead and remove it. I can just manually run those myself with `npm run format && npm run lint:fix`.

TLDR: Remove fix script, keep the rest as is.

Comment (wincent) [reply package.json]: > TLDR: Remove fix script, keep the rest as is.

Disappointed that you don't like the name, "do:not:use:or:you:will:be:fired". 😢

Comment (bryceosterhaus) [reply package.json]: 😂

First I thought, maybe call it "fix:all", but then I realized I just don't really need it and I'm always a big fan of removing code rather than adding

**Comment by wincent** [comment]
Will shipping this so that I can do the branch restructuring described in #47.
